src/routes/search.rs

```diff
@@ -76,74 +77,81 @@ impl<'v> FromFormValue<'v> for FormDateTime {
 #[post("/search", data = "<manifest_filter>")]
 pub fn search(
     manifest_filter: Option<Form<ManifestFilter>>,
-) -> Result<rocket_contrib::json::Json<Vec<Manifest>>, Error> {
-    let mut manifests = Vec::new();
-    for path in glob::glob(&format!("{}/**/*.toml", *RAVEN_REPOSITORY_PATH))? {
-        let path = path?;
-        let mut file = File::open(path)?;
-
-        // Allocate a string long enough to hold the entire file
-        let mut s = file
-            .metadata()
-            .map(|m| String::with_capacity(m.len() as usize))
-            .unwrap_or_default();
-
-        file.read_to_string(&mut s)?;
-        manifests.push(toml::from_str(&s)?);
-    }
-
-    if let Some(filter) = manifest_filter {
-        if let Some(name) = filter.name() {
-            manifests.retain(|ref x: &Manifest| x.metadata().name().contains(name));
-        }
-        if let Some(category) = filter.category() {
-            manifests.retain(|ref x: &Manifest| x.metadata().category().contains(category));
-        }
-        if let Some(description) = filter.description() {
-            manifests.retain(|ref x: &Manifest| x.metadata().description().contains(description));
-        }
-        if let Some(tags) = filter.tags() {
-            manifests.retain(|ref x: &Manifest| x.metadata().tags().contains(tags));
+) -> Result<rocket_contrib::json::Json<Vec<Manifest>>, Status> {
+    let res: Result<_, Error> = try {
+        let mut manifests = Vec::new();
+        for path in glob::glob(&format!("{}/**/*.toml", *RAVEN_REPOSITORY_PATH))? {
+            let _: Result<_, Error> = try {
```

Comment:
A comment indicating that invalid manifest are skipped would be great to explain why the `Result` returned by the `try` block isn't kept.